### PR TITLE
IAM 1031

### PR DESCRIPTION
**Pull request by shipperizer**
IAM-1031: ResourcesService implementation

- **fix: add filters to listPermissions store method**
- **chore: change name to service attribute**
- **feat: add ResourcesService**

### CODE REVIEW

**Review by nsklikas — APPROVED** [review]

File: internal/openfga/stores.go [hunk 1]
```diff
@@ -282,11 +282,92 @@ func (s *OpenFGAStore) ListPermissions(ctx context.Context, ID string, continuat
...
+	if len(errors) == 0 {
+		return permissions, tMap, nil
+	}
+
+	eMsg := ""
+
+	for n, e := range errors {
+		s.logger.Errorf(e.Error())
+		eMsg = fmt.Sprintf("%s%v - %s\n", eMsg, n, e.Error())
+	}
```
Comment (BarcoMasile) [inline internal/openfga/stores.go:360]: this looks like something we could stick in the `pool` package for reuse

File: internal/openfga/stores.go [hunk 2]
```diff
@@ -282,11 +282,92 @@ func (s *OpenFGAStore) ListPermissions(ctx context.Context, ID string, continuat
...
+// ListPermissionsWithFilters returns all the permissions associated to a specific entity
+func (s *OpenFGAStore) ListPermissionsWithFilters(ctx context.Context, ID string, opts ...ListPermissionsFiltersInterface) ([]Permission, map[string]string, error) {
+	ctx, span := s.tracer.Start(ctx, "openfga.OpenFGAStore.ListPermissionsWithFilters")
+	defer span.End()
+
+	// keep it a buffered channel, if set to unbuffered we would need a goroutine
+	// to consume from it before pushing to it
+	// https://go.dev/ref/spec#Send_statements
+	// A send on an unbuffered channel can proceed if a receiver is ready.
+	// A send on a buffered channel can proceed if there is room in the buffer
```
[the remark this answers is not in the document]
Comment (shipperizer) [reply internal/openfga/stores.go:294]: will do as part of https://github.com/canonical/identity-platform-admin-ui/issues/384

File: internal/openfga/stores.go [hunk 3]
```diff
@@ -282,11 +282,92 @@ func (s *OpenFGAStore) ListPermissions(ctx context.Context, ID string, continuat
...
+	results := make(chan *pool.Result[any], len(s.permissionTypes()))
```
[the remark this answers is not in the document]
Comment (shipperizer) [reply internal/openfga/stores.go:295]: same as above, will be tackled in https://github.com/canonical/identity-platform-admin-ui/issues/384

File: internal/openfga/stores.go (hunk 1 above)
Comment (shipperizer) [reply internal/openfga/stores.go:360]: ditto https://github.com/canonical/identity-platform-admin-ui/issues/384

**Review by BarcoMasile — APPROVED** [review]

**Pull request merged**